dipy/align/tests/test_imwarp.py

```diff
@@ -953,24 +908,16 @@ def test_em_2d_demons():
     mapping = optimizer.optimize(static, moving, None)
     m = optimizer.get_map()
     assert_equal(mapping, m)
-    energy_profile = subsample_profile(
-        optimizer.full_energy_profile, 10)
-    print(energy_profile)
-
-    if USING_VC_SSE2:
-        expected_profile = \
-            [2.50773393, 3.26942324, 1.81684393, 5.44878881, 40.0195918,
-             31.87030788, 25.15710409, 29.82206485, 196.33114499, 213.86419995]
-    elif USING_GCC_SSE2:
-        expected_profile = \
-            [2.50773393, 3.26942352, 1.8168445, 5.44879264, 40.01956373,
-             31.65616398, 32.43115903, 35.24130742, 192.89072697, 195.456909]
-
-    assert_array_almost_equal(energy_profile, np.array(expected_profile),
-                              decimal=5)
+
+    warped = mapping.transform(moving)
+    starting_energy = np.sum((static - moving)**2)
+    final_energy = np.sum((static - warped)**2)
+    reduced = 1.0 - final_energy/starting_energy
+
+    assert(reduced > 0.9)
+
 
 if __name__=='__main__':
-    test_scale_space_exceptions()
     test_optimizer_exceptions()
     test_mult_aff()
     test_diffeomorphic_map_2d()
```

Comment:
Please replace calling all the functions in the main and just use 
```run_module_suite``` and that will call all available tests in the file.